test/controllers/works_controller_test.rb

```diff
@@ -1,116 +1,468 @@
+    describe "create" do
 
+      it "will not allow a guest user access to a create work page" do
+        expect {
+          post works_path
+        }.wont_change "Work.count"
+
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
     end
 
-    it "renders 404 not_found and does not update the DB for a bogus work ID" do
+    describe "show" do
+      it "doesnt allow a guest user to view show page for work" do
+        id = works(:poodr).id
+
+        get work_path(id)
 
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
     end
-  end
 
-  describe "upvote" do
+    describe "edit" do
+      it "guest users can not edit info for an existing work ID" do
+        id = works(:poodr).id
 
-    it "redirects to the work page if no user is logged in" do
+        get edit_work_path(id)
 
-    end
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
+
+      it "##renders 404 not_found for a bogus work ID" do
+        id = -1
 
-    it "redirects to the work page after the user has logged out" do
+        get work_path(id)
+
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
+    end
 
+    describe "update" do
+      let (:work_hash) do
+        {
+          work: {
+            title: "Eat, Love, Pray",
+            creator: "Elizabeth Gilbert",
+            description: "Memoir",
+            publication_year: 2004,
+            category: "book"
+          }
+        }
+      end
+
+      it "wont allow a guest user to update existing work" do
+        id = works(:poodr).id
+
+        expect {
+          patch work_path(id), params: work_hash
+        }.wont_change "Work.count"
+
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
+
+      it "renders 404 not_found for a bogus work ID" do
+        id = -1
+
+        expect {
+          patch work_path(id)
+        }.wont_change 'Work.count'
+
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
     end
 
-    it "succeeds for a logged-in user and a fresh user-vote pair" do
+    describe "destroy" do
+      it "does not allow a guest user succeeds to delete work" do
+        id = works(:poodr).id
 
+        expect {
+          delete work_path(id)
+        }.wont_change 'Work.count'
+
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
+
+      it "renders 404 not_found and does not update the DB for a bogus work ID" do
+        id = -1
+
+        expect {
+          delete work_path(id)
+        }.wont_change 'Work.count'
+
+        must_respond_with :redirect
+        must_redirect_to root_path
+      end
     end
 
-    it "redirects to the work page if the user has already voted for that work" do
+    describe "upvote" do
+
+      it "wont let a logged out user upvote on work" do
 
+        get "/auth/github", params: {username: "dan"}
```

Comment:
This is logging in via the pre-OAuth method.  You need to mock the Omniauth hash etc.  

So use the `perform_login(user)` method in your `test_helper.rb` file.